Copying analyses notebooks

mariloubodde opened the pull request.

mariloubodde committed:
additional metadata

mariloubodde committed:
add sliding window pca notebooks

mariloubodde committed:
added pca notebooks

mariloubodde committed:
added doubleton notebooks

mariloubodde committed:
added mean fst notebooks

mariloubodde committed:
added pi and tajima d notebooks

mariloubodde committed:
added notebooks to count hets per sample

mariloubodde committed:
added notebook to count variants

mariloubodde committed:
added notebooks to plot maps

mariloubodde committed:
added gene-drive notebooks

mariloubodde committed:
updated readme

mariloubodde committed:
minor edits

mariloubodde committed:
uploaded updated version of supp1 tab2 and added supp1 tab4

mariloubodde committed:
added IBD analysis

mariloubodde committed:
add config file

mariloubodde committed:
updated access and added wattersons computations

mariloubodde committed:
replaced by updated version

mariloubodde committed:
admixture notebooks

mariloubodde committed:
updated readme

mariloubodde committed:
update way to access data and point to supp file for metadata

mariloubodde committed:
minor edits

mariloubodde committed:
added haplotype clustering notebooks

mariloubodde committed:
added H12 notebook

mariloubodde committed:
add ROH notebook

amakunin commented:
Hi @mariloubodde. I've run a selection of notebooks on datalab:
- might be worth adding a small env.yaml specifying versions of key deps (allel, zarr, malariagen_data, numpy, plotting etc) - these can be extracted from https://github.com/malariagen/mgenv/blob/v6.0.6/environment-pinned-ubuntu-latest.yml (this is for v6.0.6)
- it seems reasonable to add couple cols to suppl1_tab2 - `sample_id` that would match the one in af1 metadata (e.g., VBS10123-4954STDY7089651 for VBS10123) and short cohort codes (used in several notebooks)
- het_counts.csv contains a copy of suppl_tab2 metadata, while the analysis-specific bit is in the last few columns
- in doubletons, the paths seem to change between notebooks - `doubleton/hic30` in compute vs `hic30/doubletons_2L.tsv` in plot (also, the latter is commented to have doubletons for whole genome, not 2L)
- also, doubleton results seem to be relatively small tables - consider adding them to github, similarly to het counts
- in `compute_fst_all_cohorts`, meta column should be `median_cov` not `median_coverage`
- per-chromosome fst tsvs can be also added to github - also noted that compute notebook puts results in `fst` subdir (which needs to be created in advance - not done in compute nb), while plot notebook reads from `.`
- skipped cas9 finding code as gff downloading from vectorbase is now not trivial
- in H12 computation, `import malariagen_data; import xarray as xr` not included, `release = "1.0"` also should be added somewhere at parameters section, `geographic_cohorts = list(metadata.geographic_cohort.dropna().unique())` - add dropna to remove `nan` from cohort values,  `plots` directory should be created prior to running calibration, `geographical_cohorts` in X calibration should be replaced with `geographic_cohorts`. Also, change in keys of `funestus_windows` cohort name from `'CAR_Ombella-M-Poko'` to `'CAR_Ombella-MPoko'` is needed. Cluster shutdown missing.
- in plot H12, cohort renaming cell should be removed
- in haplotype clustering for GABA, `samples_A296S-T345S.csv` is missing, cannot do further checks
- in IBD, it seems natural to use `Fst_2L.tsv` from fst computation dir - another reason to add these fst tables to this repo
- in IBD, input fst table contains short geocohort names, while the analyses uses long geocohort names
- in compute PCA, extra_meta filename should have `_tab2` added. More importantly, `af1.add_extra_metadata(pd.read_csv("../../metadata/supp1_tab2.csv")` does not work due to sample ID column name and contents mismatch (might be easiest to add matching column to supp1_tab2). Output dir `pcas` is not created
- compute tajima's D, Pi, watterson theta nbs - `release_config_path` is incorrect, output dirs not created. `meta.query('PCA_cohort in ["EQ_central", "EQ_CAR", "EQ_east"]')` - different cohort names used in supp1_tab2, not sure if merge with af1 meta overwrites this column or not. For K/F analysis, `"../../meta/additional_sample_metadata.csv"` probably should be `supp1_tab4.tsv` - similar considerations for columns renaming. `"kiribina_folonzo_meta.tsv"` does not exist.
- plot maps - an extra pair of square brackets needed: `coords = pd.DataFrame(meta.groupby('geographic_cohort')[['latitude', 'longitude']].mean()).reset_index().rename({'index': 'geographic_cohort'})` - possible change in pandas syntax, the updated version should be backwards compatible. Also, directory for output files is not created.
- compute roh - `import malariagen_data` missing. In all sample analysis, per-chromosome ROH results are not merged properly. In single sample analysis, please remove `:` from end of last line. In the same cell, `sample_idx` does not need to be bracketed in compute functions - it's already an instance of pandas index. Cluster shutdown missing. Side note: high compute load on datalab, which requires over 6GB RAM for all-sample compute
- in `compute_sliding_window_pca.ipynb`, extra_meta filename should have `_tab2` added

mariloubodde committed:
added env information

mariloubodde committed:
changed VBS_sample_id to sample_id for compatability with malariagen_data metadata

mariloubodde committed:
metadata compatability

mariloubodde committed:
added het count results

mariloubodde committed:
vbs sample id to sample id

mariloubodde committed:
add doubleton results and fix paths

mariloubodde committed:
added fst results and fixed column names

mariloubodde committed:
updated metadata path

mariloubodde committed:
fixed small issues

mariloubodde committed:
added variant file and small fixes in nb

mariloubodde committed:
restricted Fst table to ratio of means

mariloubodde committed:
remove unnessecary code cell

mariloubodde committed:
updated metadata access pca nbs

mariloubodde committed:
added cohort shapes burkina

mariloubodde committed:
small path fixes

mariloubodde committed:
small fixes

mariloubodde committed:
small fixes

mariloubodde committed:
fixed metadata path

mariloubodde commented:
thanks @amakunin 
I have implemented all points you raised. 
- env file is added
- changed 'VBS_sample_id' to 'sample_id', which allows merging with malariagen data. cohort codes were already there under 'geographic_cohort_code'. 
- added het count results (without duplicate metadata columns) 
- fixed paths in doubleton nbs and added results (note that the plotting does read in all arms in the for loop)
- fixed requests for H12 nbs
- added sample variant info for haplotype clusters 
- IBD analyses also include Folonzo Kiribina - which is not in our main table and were computed separately. So I prefer to keep the Fst_2L.tsv file here with those additional groups in. And the cell with the long cohort names was obsolete, I have removed it.
- updated metadata access in compute PCA and rewrote to only save the results columns of the dfs
- updated path to config file in pi/tajima/watterson; removed PCA cohort based and Folonzo Kiribina analyses 
- fixed issues in plot_maps
- I actually don't use malariagen_data in roh computation, removed that cell. Now also computed fractions summarised in total.
